napari/layers/utils/layer_utils.py

```diff
@@ -807,7 +807,9 @@ def resize(
         if size < current_size:
             self.remove(range(size, current_size))
         elif size > current_size:
-            to_append = pd.concat([self._defaults] * (size - current_size))
+            to_append = self._defaults.reindex(
+                index=range(size - current_size), method='pad'
+            )
```

Comment:
I was a bit confused by this, because the index is wrong: intuitively, it should be `range(current_size, current_size + size`). But then I realized that we're discarding the index immediately after this by appending (`self.append()` uses `ignore_index=True`). So why use `reindex` in the first place?

For anyone else wondering, I played around a bit, and it turns out it *is* the faster way. However, this line can be made even faster by using `index.repeat` rather than `range`, since anyways we're discarding it:

```python
In [87]: df
Out[87]:
   a      b   c            d                                  e
0  1  thing NaN  1000.231231  <object object at 0x7fd5048aa910>

In [89]: %timeit df.reindex(index=df.index.repeat(1_000_000), method='pad')
43.5 ms ± 126 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [90]: %timeit df.reindex(index=range(10_000_000), method='pad')
408 ms ± 1.45 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [91]: %timeit df.loc[np.tile(0, 10_000_000)]
1.06 s ± 8.53 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [92]: %timeit df.iloc[np.tile(0, 10_000_000)]
484 ms ± 9.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
Actually, `np.tile` and `np.repeat` seem comparable to `repeat`:

```
In [93]: %timeit df.reindex(index=np.tile(0, 1_000_000), method='pad')
43.4 ms ± 305 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [94]: %timeit df.reindex(index=np.repeat(0, 1_000_000), method='pad')
43.2 ms ± 392 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
Isn't the main difference in the times here explained by using `1_000_000` vs. `10_000_000`? Except for `loc`, which is unsurprisingly slower than `iloc` et al. If I run three of the options with `1_000_000`, I get:

```python
In [38]: %timeit df.reindex(index=range(1_000_000), method='pad')
24.7 ms ± 314 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [39]: %timeit df.iloc[np.zeros(1_000_000)]
27.9 ms ± 856 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [40]: %timeit df.loc[np.zeros(1_000_000)]
35.5 ms ± 792 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

But point taken that the index here isn't useful because we're going to use `ignore_index=True` in `append`, so using `reindex` is a little confusing. Therefore, I prefer the `iloc` form as it most clearly states what we want here (a dataframe that repeats the single default row) and is almost as fast as `reindex`, so I used that instead (i.e. `df.iloc[np.zeros(1_000_000)]`) - what do you think?

Reply:
> Isn't the main difference in the times here explained by using `1_000_000` vs. `10_000_000`? Except for `loc`, which is unsurprisingly slower than `iloc` et al. If I run some of the options with `1_000_000`, I get:

LOL, sorry, I've done goofed :facepalm: I was testing higher and higher number to see how it scaled, and things started getting worse with bigger numbers, so I assumed it made sense; apparently it was a typo :P

> Therefore, I prefer the iloc form as it most clearly states what we want here (a dataframe that repeats the single default row) and is almost as fast as reindex - what do you think?

Agreed, so for clarity's sake I would go with the more commonly-known functions:

```
df.loc[np.zeros(1_000_000)]
```

Other than that, LGTM!

Reply:
Btw, a more thorough (and correct xD) benchmark just for completeness:

```python
In [5]: df
Out[5]:
   a      b   c            d                                  e
0  1  thing NaN  1000.231231  <object object at 0x7f65387c8520>

In [6]: %timeit df.reindex(index=np.repeat(0, 100_000_000), method='pad')
5.17 s ± 206 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: %timeit df.reindex(index=range(100_000_000), method='pad')
4.77 s ± 27 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [8]: %timeit df.reindex(index=np.tile(0, 100_000_000), method='pad')
4.88 s ± 286 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [9]: %timeit df.reindex(index=df.index.repeat(100_000_000), method='pad')
4.8 s ± 140 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [10]: %timeit df.iloc[np.tile(0, 100_000_000)]
5.16 s ± 28 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [11]: %timeit df.iloc[np.repeat(0, 100_000_000)]
5.13 s ± 31.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [12]: %timeit df.iloc[df.index.repeat(100_000_000)]
5.4 s ± 521 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [13]: %timeit df.iloc[np.zeros(100_000_000)]
5.1 s ± 43.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Reply:
I stuck with `iloc` because it's a decent bit faster than `loc`, I wouldn't describe it as an uncommon function [given its prominence in the docs](https://pandas.pydata.org/docs/user_guide/indexing.html#different-choices-for-indexing), and there's only one row in the default dataframe.

Reply:
(sorry, I meant `iloc` as well, *another* typo. It's not my day xD)